xbitinfo/xbitinfo.py

```diff
@@ -35,6 +36,46 @@ def get_user_input():
     return args
 
 
+def get_bit_coords(dtype_size):
+    """Get coordinates for bits assuming float dtypes."""
+    if dtype_size == 16:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 6)]
+            + [f"m{int(i-8)}" for i in range(6, 16)]
+        )
+    elif dtype_size == 32:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 9)]
+            + [f"m{int(i-8)}" for i in range(9, 32)]
+        )
+    elif dtype_size == 64:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 12)]
+            + [f"m{int(i-11)}" for i in range(12, 64)]
+        )
+    else:
+        raise ValueError(f"dtype of size {dtype_size} neither known nor implemented.")
+    return coords
+
+
+def dict_to_dataset(info_per_bit):
+    """Convert keepbits dictionary to dataset."""
+    dsb = xr.Dataset()
+    for v in info_per_bit.keys():
+        dtype_size = len(info_per_bit[v])
+        dim_name = f"bit{dtype_size}"
+        dsb[v] = xr.DataArray(
+            info_per_bit[v],
+            dims=[dim_name],
+            coords={dim_name: get_bit_coords(dtype_size)},
+            name=v,
+        ).astype("float32")
```

Comment:
had to change to float32 because float16 limits the cdf to 4 aftercommadigits  and `inflevel=0.9999` and more close to 100% wont work